src/virtual_machine/register.rs

```diff
@@ -50,6 +50,16 @@ impl Registers {
     }
 }
 
+impl ToString for Registers {
+    fn to_string(&self) -> String {
+        self.registers
+            .iter()
+            .fold(String::new(), |accumulator, register| {
+                accumulator + &format!("{register:#018b}::")
+            })
+    }
+}
+
```

Comment:
The Rust documentation discourages the use of `ToString`, encouraging the use of `Display` instead, through which you get `.to_string()` for free.

https://doc.rust-lang.org/std/string/trait.ToString.html

> This trait is automatically implemented for any type which implements the [Display](https://doc.rust-lang.org/std/fmt/trait.Display.html) trait. As such, ToString shouldn’t be implemented directly: [Display](https://doc.rust-lang.org/std/fmt/trait.Display.html) should be implemented instead, and you get the ToString implementation for free.